app/forms/articles/destroy_form.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Articles
+  class DestroyForm < Articles::BaseForm
+    def save
+      ActiveRecord::Base.transaction do
+        save_article_tags!
```

Comment:
it's not the save action - it's assign. you can remove this line of code and simply use `@article.tags` in `check_and_destroy_tags!`

Reply:
I can't, because when article is deleted from the DB in 8th line - I won't be able to get its tags. And the logic breaks down. 
Also, I can't process tags first until the article is destroyed. So I save these tags in a buffer variable.
Not "save", but "asign".